client/components/data/query-canonical-theme/index.jsx

```diff
@@ -0,0 +1,37 @@
+/**
+ * External dependencies
+ */
+import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import QueryTheme from 'components/data/query-theme';
+import { isWpcomTheme, isWporgTheme } from 'state/themes/selectors';
+
+const QueryCanonicalTheme = ( { siteId, themeId, isWpcom, isWporg } ) => (
+	<div>
+		<QueryTheme themeId={ themeId } siteId="wpcom" />
```

Comment:
I don't think that it is worth investigating given the fact that it works the way it is now and that there will be no re-render.